src/core/datastore/datastore.doc.js

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable */
```

Comment:
What is the motivation behind separating this in a `.doc.js` file? Do we have such files for other classes?

Reply:
The motivation is that in the DevCenter DataStore guides we discuss `DataStoreType` and not different data store classes. If I was to document the classes, it would be confusing since these classes are not available publicly. By using `datastore.doc.js` I can document a data store without requiring the implementation to fit into the way esdoc needs it to be to follow the way we discuss it in our guides.

This is the only area in the SDK where this needs to be done at the moment and we do not had a `.doc.js` anywhere else for other classes. This is the pattern would probably use if this scenario was encountered elsewhere ins the SDK.